src/chainstate/stacks/db/blocks.rs

```diff
@@ -5280,21 +5279,37 @@ impl StacksChainState {
         // epoch defined by this miner.
         clarity_tx.reset_cost(ExecutionCost::zero());
 
-        debug!("Evaluating block with epoch = {}", evaluated_epoch);
+        // is this stacks block the first of a new epoch?
+        let (applied_epoch_transition, mut tx_receipts) =
+            StacksChainState::process_epoch_transition(&mut clarity_tx, burn_tip_height)?;
+
+        let evaluated_epoch = clarity_tx.get_epoch();
+
         if evaluated_epoch >= StacksEpochId::Epoch21 {
-            Self::check_and_handle_reward_start(
+            let events = Self::check_and_handle_reward_start(
                 burn_tip_height.into(),
                 burn_dbconn,
                 sortition_dbconn,
                 &mut clarity_tx,
                 chain_tip,
                 &parent_sortition_id,
             )?;
-        }
 
-        // is this stacks block the first of a new epoch?
-        let (applied_epoch_transition, mut tx_receipts) =
-            StacksChainState::process_epoch_transition(&mut clarity_tx, burn_tip_height)?;
+            let unlock_tx = StacksTransactionReceipt {
+                transaction: TransactionOrigin::NetworkProtocol,
+                events,
+                post_condition_aborted: false,
+                result: Value::okay_true(),
+                stx_burned: 0,
+                contract_analysis: None,
+                execution_cost: ExecutionCost::zero(),
+                microblock_header: None,
+                tx_index: 0,
+                vm_error: None,
+            };
```

Comment:
This looks great. Could you share an example of how this synthetic tx will look like in node messages, though? e.g. will it have the value of `0x00` for the `raw_tx` entry?

Reply:
Yes, the txid field will be equal to "NetworkProtocol", and and the raw_tx field will be set to "00".

Reply:
On the API, we use the `tx_id` as a unique key in combination with the `index_block_hash` and `microblock_hash`... I'm afraid that if we use the same string for all of these transactions, conflicts would arise if two addresses have an unlock tx in the same block.

Is it possible for `tx_id` to be a (random?) hex value like all others? Or what would the implications be?